src/tests/Dashboard/container/DictionaryDisplay.test.jsx

```diff
@@ -114,17 +114,19 @@ describe('DictionaryDisplay', () => {
       onSubmit: jest.fn(),
       organizations: [organizations],
       history: { push: jest.fn() },
-      searchValue: '',
     };
-    const wrapper = mount(<MemoryRouter>
-      <DictionaryDisplay {...props} />
-    </MemoryRouter>);
+    const wrapper = mount(<DictionaryDisplay {...props} />);
     const event = { target: { name: 'searchInput', value: 'openmrs' } };
     const event2 = { target: { name: 'searchInput', value: '', type: 'checkbox' } };
     wrapper.find('#search').simulate('change', event);
     wrapper.find('#search').simulate('change', event2);
-    wrapper.find('#submit-search-form').simulate('submit', {
-      preventDefault: () => {},
+    wrapper.setState({ searchInput: 'test' }, () => {
+      wrapper.update();
+      wrapper.find('#submit-search-form').simulate('submit', { preventDefault: jest.fn() });
+    });
+    wrapper.setState({ searchInput: 'ka' }, () => {
```

Comment:
Are you asserting anywhere for this test about expected output?

Reply:
@dkayiwa I've fixed this

Reply:
Don't you expect specific results?

Reply:
The data/results tests are for `Actions` and `Reducers` because they are the ones that make a call to the backend, and change Redux state respectively.

Reply:
I thought you would point me to the exact lines of the test you are referring to.

Reply:
`dictionaryAction.test.js`

```
it('should return an array of dictionaries on search', () => {
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: [dictionaries],
      });
    });

    const expectedActions = [
      { type: IS_FETCHING, payload: true },
      { type: FETCHING_DICTIONARIES, payload: [dictionaries] },
      { type: IS_FETCHING, payload: false },
    ];

    const store = mockStore({ payload: {} });
    return store.dispatch(
      searchDictionaries('', 1000, 1, 'sortAsc=name', 'verbose=true'),
    ).then(() => {
      expect(store.getActions()).toEqual(expectedActions);
    });
  });
```

Partial search is not separate from how search was being done. It just adds an asterisk to ALL search queries so that whether it's a full word or not, it is still searched for.

Reply:
If a bug is introduced and results are only returned for full words instead of part of the word, do we have a test which will fail?

Reply:
I've added tests which will definitely fail if the asterisk is missing.
```
expect(wrapper.state().searchInput).toEqual('test');
expect(searchDictionaries).toHaveBeenCalledWith(`${wrapper.state().searchInput}*`);
```
Without it, the backend does not return results for partial search.

Reply:
Can you make the test expect specific values returned according to the mock data?

Reply:
> Can you make the test expect specific values according to the mock data?

That's exactly what the test in Actions (displayed above) is doing. It expects a list of dictionaries to be passed as a payload to the action if that list was returned from the request.

The same test covers both partial search and word search because they work exactly the same way. The only difference would be an asterisk which I have tested for to ensure that it always exists in the search query.

Reply:
It expects a list containing how many dictionaries?

Reply:
Just one. It's defined as a mock in `dictionaries.js`

Reply:
Can you make it expect 3?

Reply:
Yes. But the result won't change. What is passed in the `request` section below is what will be expected in the `expectedActions` because it's not actually making a call to the backend. It's just a mock.
```
request.respondWith({
        status: 200,
        response: [dictionaries],
      });
```

Would you still like me to make it 3?

Reply:
yes

Reply:
I've done so

Reply:
Can you point me to the line which expects them to be 3?

Reply:
There's no such line.
Please note that this Action takes effect AFTER data is received from the backend. It's the backend that performs the actual search. Therefore what we test for is whether what is received is actually passed through the action. It will not have any logic that sifts through the results.